host_cli/main.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+// TODO: Eventually the .dive file support and the raw data support in `cli/` will be migrated here
+// and the old cli will be deprecated
+
+#include <filesystem>
+#include <string>
+
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/flags/usage.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_format.h"
+
+#include "data_core_wrapper.h"
+
+constexpr std::array kAllowedInputFileExtensions = { ".gfxr" };
+
+ABSL_FLAG(bool, version_info, false, "Shows the version of Dive host tools and quits");
+ABSL_FLAG(std::string,
+          input_file_path,
+          "",
+          "If specified, this file is loaded with the type is inferred from the extension. "
+          "Supported extensions: .gfxr");
+ABSL_FLAG(std::string,
+          output_gfxr_path,
+          "",
+          "If specified, a new .gfxr file will be generated from the original file "
+          "(--input_file_path) and any specified modifications");
+
+absl::Status ValidateFlags()
+{
+    std::string input_file_ext = "";
+
+    std::string input_file_path = absl::GetFlag(FLAGS_input_file_path);
+    if (!input_file_path.empty())
+    {
+        std::filesystem::path input_fp = input_file_path;
+        std::string           ext = input_fp.extension().string();
+        if ((ext == "") || std::find(kAllowedInputFileExtensions.begin(),
+                                     kAllowedInputFileExtensions.end(),
+                                     ext) == kAllowedInputFileExtensions.end())
+        {
+            return absl::InvalidArgumentError(
+            absl::StrFormat("unrecognizable extension provided for --input_file_path: %s", ext));
+        }
+        input_file_ext = ext;
+    }
+
+    std::string output_gfxr_path = absl::GetFlag(FLAGS_output_gfxr_path);
+    if (output_gfxr_path != "")
```

Comment:
consider using `!empty()` here too
```suggestion
    if (!output_gfxr_path.empty())
```

Reply:
Fixed